### DISCUSSION TITLE
[질문] soft delete 적용 관련 질문

### DISCUSSION THREAD

**Discussion by kang20** [Kumoh-talk/kumoh-talk-Backend#60] [Q&A]
![image](https://github.com/user-attachments/assets/fe6460cb-6040-49e3-b624-cf2affb5a83e)

저희 soft delete 를 적용하는 이유가 물리적인 삭제시 완전히 삭제된다는 점 때문이라서 Board에는 soft delete 를 적용하지만 
BoardCategroy 같은 다대다 관계에 중계 테이블에도 soft delete를 적용할 필요가 있을까요?

**Comment by kimgunwooo** [comment]
삭제된 글에 대해서도 카테고리별로 조회가 필요한 경우가 있다면 필요하지 않을까 싶긴 하네요.

근데 적용 안할 이유가 있을까요??
추후 오래된 데이터에 대해서 삭제처리할때 귀찮아질까봐서 인가요?

**Reply by kang20** [reply]
중계 테이블을 삭제 안하고 보관해야할 필요가 있을까 해서 그렇습니다

**Reply by kimgunwooo** [reply]
제가 생각하기에는 soft delete 적용한 이상 남겨둬도 괜찮지 않을까 싶네요.
그렇게 많이 쌓일 것 같지도 않고, 정보가 조금이라도 더 남아있는게 좋지 않을까 싶긴하네요..
